isoutils/isoutils.go

```diff
@@ -0,0 +1,517 @@
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: GPL-3.0-only
+
+package isoutils
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"syscall"
+	"text/template"
+
+	"github.com/clearlinux/clr-installer/args"
+	"github.com/clearlinux/clr-installer/cmd"
+	"github.com/clearlinux/clr-installer/log"
+	"github.com/clearlinux/clr-installer/model"
+	"github.com/clearlinux/clr-installer/progress"
+	"github.com/clearlinux/clr-installer/swupd"
+)
+
+type Paths int
+
+const (
+	clr_efi     Paths = iota + 0
+	clr_img_efi       // Location to mount the EFI partition in the passed-in img file
+	clr_initrd
+	clr_rootfs
+	clr_cdroot
+)
+
+var (
+	tmpPaths = make([]string, 5)
+)
+
+func mkTmpDirs() error {
+	msg := "Creating directory trees"
+	prg := progress.NewLoop(msg)
+	log.Info(msg)
+	var err error
+
+	tmpPaths[clr_efi], err = ioutil.TempDir("", "clr_efi-")
+	if err != nil {
+		prg.Failure()
+		return (err)
+	}
+	tmpPaths[clr_initrd], err = ioutil.TempDir("", "clr_initrd-")
+	if err != nil {
+		prg.Failure()
+		return (err)
+	}
+	tmpPaths[clr_cdroot], err = ioutil.TempDir("", "clr_cdroot-")
+	if err != nil {
+		prg.Failure()
+		return (err)
+	}
+
+	/* Create specific directories for the new cd's root */
+	for _, d := range []string{
+		tmpPaths[clr_cdroot] + "/isolinux",
+		tmpPaths[clr_cdroot] + "/EFI",
+		tmpPaths[clr_cdroot] + "/images",
+		tmpPaths[clr_cdroot] + "/kernel",
+	} {
+		if _, err := os.Stat(d); os.IsNotExist(err) {
+			os.Mkdir(d, os.ModePerm)
+		}
+	}
+
+	prg.Success()
+	return err
+}
+
+func mkRootfs() error {
+	msg := "making rootfs squashfs"
+	prg := progress.NewLoop(msg)
+	log.Info(msg)
+
+	/* TODO: This takes a long time to run, it'd be nice to see it's output as it's running */
+	args := []string{
+		"mksquashfs",
+		tmpPaths[clr_rootfs],
+		tmpPaths[clr_cdroot] + "/images/rootfs.img",
+		"-comp",
+		"lzo",
+		"-wildcards",
```

Comment:
We can drop wildcards as we do not need them when only excluding top-level directories.

Reply:
Done.